server/src/test/java/org/elasticsearch/gateway/MetaStateServiceTests.java

```diff
@@ -21,83 +21,163 @@
 import org.elasticsearch.Version;
 import org.elasticsearch.cluster.metadata.IndexMetaData;
 import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.metadata.MetaState;
+import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.NodeEnvironment;
-import org.elasticsearch.index.Index;
 import org.elasticsearch.test.ESTestCase;
 
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 
 public class MetaStateServiceTests extends ESTestCase {
-    private static Settings indexSettings = Settings.builder()
-            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
-            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .build();
+    private static Settings indexSettings() {
```

Comment:
I think I'd rather have a method `IndexMetaData indexMetaData(String indexName)` - it's slightly surprising that this method returns different settings each time it's called, but it's only ever used in the context of `IndexMetaData.builder(indexName).settings(indexSettings()).build();`

Reply:
Good catch!

Reply:
https://github.com/elastic/elasticsearch/pull/35199/commits/6d178c5fb951a1ac134ab0353ef1cd79e764b093